daemon/src/monitor.rs

```diff
@@ -33,26 +34,19 @@ pub struct MonitorParams {
 
 pub struct Sync;
 
-pub struct Actor<T, C = bdk::electrum_client::Client>
-where
-    T: xtra::Actor,
-{
+pub struct Actor<C = bdk::electrum_client::Client> {
     cfds: HashMap<OrderId, MonitorParams>,
-    cfd_actor_addr: xtra::Address<T>,
-
+    event_channel: Box<dyn StrongMessageChannel<Event>>,
     client: C,
     latest_block_height: BlockHeight,
     current_status: BTreeMap<(Txid, Script), ScriptStatus>,
     awaiting_status: HashMap<(Txid, Script), Vec<(ScriptStatus, Event)>>,
 }
 
-impl<T> Actor<T, bdk::electrum_client::Client>
-where
-    T: xtra::Actor + xtra::Handler<Event>,
-{
+impl Actor<bdk::electrum_client::Client> {
     pub async fn new(
         electrum_rpc_url: &str,
-        cfd_actor_addr: xtra::Address<T>,
+        event_channel: Box<dyn StrongMessageChannel<Event>>,
```

Comment:
Perhaps change this to `impl StrongMessageChannel<Event>`, then you can pass in an address and do the boxing inside.